kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,831 @@
+        self.ctx = None
+
+    def _init_toolbar(self):
+        basedir = os.path.join(rcParams['datapath'], 'images')
+        actionview = ActionView()
+        actionprevious = ActionPrevious(title="Navigation", with_previous=False)
+        actionoverflow = ActionOverflow()
+        actionview.add_widget(actionprevious)
+        actionview.add_widget(actionoverflow)
+        actionview.use_separator = True
+        self.actionbar.add_widget(actionview)
+        for text, tooltip_text, image_file, callback in self.toolitems:
+            if text is None:
+                actionview.add_widget(ActionSeparator())
+                continue
+            fname = os.path.join(basedir, image_file + '.png')
+            if text in ['Pan', 'Zoom']:
+                action_button = ActionToggleButton(text=text, icon=fname)
+            else:
+                action_button = ActionButton(text=text, icon=fname)
+            action_button.bind(on_press=getattr(self, callback))
+            actionview.add_widget(action_button)
+
+    def configure_subplots(self):
+        pass
+
+    def dismiss_popup(self):
+        self._popup.dismiss()
+
+    def show_save(self):
+        content = SaveDialog(save=self.save, cancel=self.dismiss_popup)
+        self._popup = Popup(title="Save file", content=content,
+                            size_hint=(0.9, 0.9))
+        self._popup.open()
+
+    def save(self, path, filename):
+        with open(os.path.join(path, filename), 'w') as stream:
+            stream.write(self.text_input.text)
+        self.dismiss_popup()
+
+    def save_figure(self, *args):
+        self.show_save()
+
+
+class GraphicsContextKivy(GraphicsContextBase):
+    """
+    The graphics context provides the color, line styles, etc... See the gtk
+    and postscript backends for examples of mapping the graphics context
+    attributes (cap styles, join styles, line widths, colors) to a particular
+    backend. In GTK this is done by wrapping a gtk.gdk.GC object and
+    forwarding the appropriate calls to it using a dictionary mapping styles
+    to gdk constants. In Postscript, all the work is done by the renderer,
+    mapping line styles to postscript calls.
+
+    If it's more appropriate to do the mapping at the renderer level (as in
+    the postscript backend), you don't need to override any of the GC methods.
+    If it's more appropriate to wrap an instance (as in the GTK backend) and
+    do the mapping here, you'll need to override several of the setter
+    methods.
+
+    The base GraphicsContext stores colors as a RGB tuple on the unit
+    interval, e.g., (0.5, 0.0, 1.0). You may need to map this to colors
+    appropriate for your backend.
+    """
+
+    _capd = {
+        'butt': 'square',
+        'projecting': 'square',
+        'round': 'round',
+    }
+
+    def __init__(self, renderer):
+        #super(GraphicsContextBase, self).__init__()
+        GraphicsContextBase.__init__(self)
```

Comment:
When you hit a bug, you need to fix it right there or open a issue for it. You cannot leave stuff like this in the code.